src/screens/map-screen/map-screen.component.tsx

```diff
@@ -191,9 +200,61 @@ const MapScreen = () => {
     setCurrentRegion(getCampusById(CampusId.SGW).region);
   }, []);
 
+  /**
+   *  Set the value of endLocation or initial location depending
+   *  on which input the user is focued
+   * @param poi
+   */
+  const setMarkerLocation = (poi: POI | null) => {
+    if (endLocationFocused) {
+      setEndLocation(poi);
+    } else {
+      setStartLocation(poi);
+    }
+  };
+
+  const setUserLocation = () => {
+    if (!currentLocation) {
+      getCurrentLocationAsync()
+        .then(location => {
+          setCurrentLocation({
+            latitude: location.coords.latitude,
+            longitude: location.coords.latitude
+          });
+        })
+        .catch();
+    }
+  };
+
+  let search;
+  if (travelState === TravelState.PLANNING) {
+    search = (
+      <OmniboxDirections
+        endLocation={endLocation}
+        currentLocation={currentLocation}
+        setEndLocation={setEndLocation}
+        setStartLocation={setStartLocation}
+        startLocation={startLocation}
+        queryText={queryText}
```

Comment:
Good point. A remnant of refactored code

Reply:
After refactoring I was having trouble with getting the tests to work since this function involves state. At the moment I've kept it as is, if you're available to help with this let me know.